### ISSUE TITLE
renameClassNamed:to: incorrectly updates classExtension method data structures

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/Rowan#535]
Originally reported as [a Jadeite bug](https://github.com/GemTalk/Jadeite/issues/643), the classExtension registry corruption was incorrectly associated with issue #490. When tests were run against the proposed fix for #490 in commit b858b8c, some of the #490 tests were still failing:
```
3.2.15 and 3.5.0
  RwRowanSample4Test>>#testIssue490_rename_package_move_newClassVersion_newProject_1
```
while the explicit #renameClassNamed:to: test was passing:
```
3.2.15 and 3.5.0
  RwBrowserToolApiTest>>testRenameClass_Issue_490
```
Which all implies that the the proposed fix for #490 in commit b858b8c does not completely address the issues in #490, which include move a the new version of a class to a different package (something RwBrowserToolApiTest>>testRenameClass_Issue_490 does not do).

**Comment by dalehenrich** [comment]
An additional variant of this bug shows up after the proposed fix for #495 (2d546c0d):
```
RwRowanSample9Test>>testIssue_spec_002_to_003
---------------------------------------------
==============Auditing  package RowanSample9-Core
---------------RowanSample9Class1 class class>>Missing loaded method>>bar
==============Auditing  package RowanSample9-Extensions
'RowanSample9Class1 #*rowansample9-extensions'->' loaded extension class (137496577) not latest version of class (137492993) '
---------------------------------------------
```
the #495 fix involves handling extension method extension moves "correctly", but the logic is not quite complete when new class versions are involved ... which is why `RwRowanSample9Test>>testIssue_spec_002_to_003` started failing ...

**Comment by dalehenrich** [comment]
The current variant `RwRowanSample9Test>>testIssue_spec_002_to_003` does not involve a new version of the class --- but the loadedExtension data structure is corrupted in a similar fashio ...
